### ISSUE TITLE
Unknown api exception 4 while creating Firebase Short Dynamic link

### ISSUE THREAD

**Issue by febimfelix** [firebase/quickstart-android#672]
* Android device: _____Samsung & Motorola
  * Android OS version: _____Both Nougat & Marshmallow
  * Google Play Services version: _____15.0.1
  * Firebase/Play Services SDK version: _____16.0.3
  
This is the error i am getting when i try to create a short dynamic link.
com.google.android.gms.common.api.ApiException: 8: 

```
FirebaseDynamicLinks.getInstance().createDynamicLink()
                .setLink(Uri.parse(userUrlLink))
                .setDynamicLinkDomain(getResources().getString(R.string.firebase_dynamic_link_domain))
                .setAndroidParameters(new DynamicLink.AndroidParameters.Builder().build())
                .setIosParameters(new DynamicLink.IosParameters.Builder(getResources().getString(R.string.ios_app_identifier))
                        .setAppStoreId(getResources().getString(R.string.ios_app_store_id)).build())
                .buildShortDynamicLink()
                .addOnCompleteListener(new OnCompleteListener<ShortDynamicLink>() {
                    @Override
                    public void onComplete(@NonNull Task<ShortDynamicLink> task) {
                        try {
                            if(!task.isSuccessful()) {
                                Log.e("DYNAMIC LINK", task.getException() == null ? "" : task.getException().getMessage());
                            }
                            if (task.isSuccessful())
                                mDynamicLinkToShare = URLDecoder.decode(Uri.decode(task.getResult().getShortLink().toString()), "UTF-8");
                        } catch (UnsupportedEncodingException e) {
                            e.printStackTrace();
                        }
                    }
                });
```

**Comment by samtstern** [comment]
The code `8` is equal to `CommonStatusCodes.INTERNAL_ERROR`
https://developers.google.com/android/reference/com/google/android/gms/common/api/CommonStatusCodes.html#INTERNAL_ERROR

Can you reproduce this with the sample app in the repo without changing any code, or is there a particular option you're setting that causes this issue?

**Comment by febimfelix** [comment]
Hello,

I downgrade the Firebase SDK version to 15.0.1 and that solved the issue. Why this version conflict between Firebase and Google Play Services still exists?

**Comment by febimfelix** [comment]
Fixed version

```
implementation 'com.google.android.gms:play-services-plus:15.0.1'
    implementation 'com.google.android.gms:play-services-auth:15.0.1'
    implementation 'com.google.android.gms:play-services-location:15.0.1'
    implementation 'com.google.android.gms:play-services-maps:15.0.1'
    implementation 'com.google.android.gms:play-services-places:15.0.1'
    implementation 'com.google.maps.android:android-maps-utils:0.5'

    implementation 'com.google.firebase:firebase-core:15.0.2'
    implementation 'com.google.firebase:firebase-messaging:15.0.2'
    implementation 'com.google.firebase:firebase-invites:15.0.1'
    /*implementation 'com.crashlytics.sdk.android:crashlytics:2.9.2'*/
    implementation 'com.google.android.gms:play-services-tagmanager:15.0.2'
```

Broken Version

```
implementation 'com.google.android.gms:play-services-auth:16.0.0'
    implementation 'com.google.android.gms:play-services-plus:15.0.1'
    implementation 'com.google.android.gms:play-services-location:15.0.1'
    implementation 'com.google.android.gms:play-services-maps:15.0.1'
    implementation 'com.google.android.gms:play-services-places:15.0.1'
    implementation 'com.google.maps.android:android-maps-utils:0.5'

    implementation 'com.google.firebase:firebase-core:16.0.3'
    implementation 'com.google.firebase:firebase-messaging:17.3.1'
    implementation 'com.google.firebase:firebase-invites:16.0.3'
    /*implementation 'com.crashlytics.sdk.android:crashlytics:2.9.2'*/
    implementation 'com.google.android.gms:play-services-tagmanager:16.0.3'
```

**Comment by PaulWoitaschek** [comment]
100% reproducible.

## Works:
com.google.firebase:firebase-dynamic-links:16.0.1
## Api Exception
16.1.1

**Comment by PaulWoitaschek** [comment]
Also see:
https://stackoverflow.com/questions/52152116/short-dynamic-link-error-com-google-android-gms-common-api-apiexception-8

**Comment by YogevUzan** [comment]
Are there news regarding this issue?  I tried downgrading to

> com.google.firebase:firebase-dynamic-links:16.0.1


but I keep getting this error:

> The library com.google.android.gms:play-services-measurement-base is being requested by various other libraries at [[15.0.2,15.0.2]], but resolves to 16.0.3.

I couldn't figure out which one is using the library.

my gradle:

>     implementation 'com.google.android.gms:play-services-analytics:16.0.4'
    implementation 'com.google.firebase:firebase-perf:16.1.2'
    implementation 'com.google.firebase:firebase-core:16.0.4'
    implementation 'com.firebaseui:firebase-ui-storage:3.2.2'
    implementation 'com.firebaseui:firebase-ui-auth:4.0.0'
    implementation 'com.firebaseui:firebase-ui-database:3.3.1'
    implementation 'com.google.firebase:firebase-ads:17.0.0'
    implementation 'com.firebaseui:firebase-ui-firestore:4.2.0'
    implementation 'com.google.firebase:firebase-auth:16.0.4'
    implementation 'com.google.firebase:firebase-database:16.0.3'
    implementation 'com.google.firebase:firebase-storage:16.0.3'
    implementation 'com.google.firebase:firebase-core:16.0.4'
    implementation 'com.google.firebase:firebase-messaging:17.3.3'
    implementation 'com.google.firebase:firebase-invites:16.0.4'

**Comment by samtstern** [comment]
Sorry I never got back to this.

As you can see in the SO answer below, this is a known issue:
https://stackoverflow.com/a/52157428/324977

For now you have to downgrade or follow the steps there.
